### ISSUE TITLE
Upgrade to SQLite 3.34.1

### ISSUE THREAD

**Issue by thomasnield** [pawelsalawa/sqlitestudio#3470]
Can we please get an upgrade to the latest version of SQLite released? There's a lot of new features added to SQLite since then, primarily [windowing functions](https://www.sqlite.org/releaselog/3_25_0.html).

**Comment by pawelsalawa** [comment]
Apart from upgrading SQLite version SQLiteStudio should start supporting window functions. This will be tracked by a separate issue (#3471), aimed for 3.3.0 (while SQLite version upgrade will be done within 3.2.2).

Upgrading to 3.25.0 or newer requires modifications in TableModifier class, as 3.25 fixed few things in ALTER TABLE statement and referencing objects:

> Enhancements the ALTER TABLE command:
> Add support for renaming columns within a table using ALTER TABLE table RENAME COLUMN oldname TO newname.
> Fix table rename feature so that it also updates references to the renamed table in triggers and views.

A quick way to handle it is "PRAGMA legacy_alter_table" if there is not enough time to do it properly in 3.2.2.

**Comment by pawelsalawa** [comment]
Depends on #3726

**Comment by pawelsalawa** [comment]
Upgrade to 3.32

**Comment by pawelsalawa** [comment]
Please don't close it. It's not fully upgraded yet (syntax definitions, AST, and stuff like that needs update too before closing).

**Comment by thomasnield** [comment]
Oops sorry. Thanks for getting to this!

**Comment by roy-marquez** [comment]
Yes please, it would be great to have support for "ALTER TABLE table RENAME COLUMN oldname TO newname" statement. For SQLite 3.32 in general.
Thank you!.

**Comment by pawelsalawa** [comment]
Notes for myself.

What's left for the SQLite version upgrade:
* Library upgrade in plugins (SQLCipher, wxSQLite, System.Data.Sqlite)
* Column Dialog - add the GENERATED constraint support
* To test and fix any bugs in GridView/FormView regarding the new GENERATED columns
* Syntax definition & UI enhancements for Window Functions
* The TableModifier needs update regarding enhanced `ALTER TABLE RENAME` - as described in the comment above.
* To go through SQLite release changelog since 3.24 and see what other syntax changes were introduced - then update this list.
